src/tests/test_magic_cmd.py

```diff
@@ -416,9 +416,25 @@ def test_test_error(ip, cell, error_type, error_message):
     assert str(out.error_in_exec) == error_message
 
 
-def test_snippet(ip_snippets):
-    out = ip_snippets.run_cell("%sqlcmd snippets").result
-    assert "high_price, high_price_a, high_price_b" in out
+@pytest.mark.parametrize(
+    "precmd, valid_snippets, invalid_snippet",
+    [
+        (None, ["high_price", "high_price_a", "high_price_b"], "invalid"),
+        (
+            "%sqlcmd snippets -d high_price_a",
+            ["high_price", "high_price_b"],
+            "high_price_a",
+        ),
+        ("%sqlcmd snippets -A high_price", ["No snippets stored"], "high_price"),
+    ],
+)
+def test_snippet(ip_snippets, precmd, valid_snippets, invalid_snippet):
```

Comment:
we're not testing the returned object. when there are no snippets, you can check that you get a Message. when there is a snippet, you can test you get a Table, this way we'll ensure there is an HTML representation and a plain-text representation